### PR TITLE
🎉 Source Instagram: Improve, refactor `STATE` management

### LINKED ISSUE #22929: Source Instagram: user_lifetime_insights stream fails due to KeyError: 'end_time'
## Environment
- **Source Connector and version**: Instagram - 1.0.1
- **Step where error happened**:  Sync job 

## Current Behavior
When a connection is created and data sync is triggered. While streaming ```user_lifetime_insights``` the sync job fails because the stream records are expected to have end_time field present. But  as per the [documentation](https://developers.facebook.com/docs/instagram-api/guides/insights/), lifetime insights wouldn't have that field. Hence, its throwing an error and job fails.

## Expected Behavior
That field should not be parsed from the graph api response, instead the datetime at which the record was fetched, that can be used as end_time.

## Logs
Yes. 
[workspace_logs_117_txt.txt](https://github.com/airbytehq/airbyte/files/10723482/workspace_logs_117_txt.txt)

## Steps to Reproduce
1. Create a connection for instagram ads.
2. Make sure user_lifetime_insights stream is selected for sync.
3. Sync the data. 
4. You will see the sync failing when it tries to sync the user_lifetime_insights stream.

## Are you willing to submit a PR?
Yes. 
https://github.com/airbytehq/airbyte/pull/22928

### LINKED ISSUE #24697: Source Instagram: incremental mode deprecated
## Environment
- **Airbyte version**:  0.42.0
- **OS Version / Instance**: Widows10 with WSL
- **Deployment**: Docker env
- **Source Connector and version**: source-instagram 
- **Destination Connector and version**: DB Postgres
- **Step where error happened**: no error

## Current Behavior
UserInsights stream contains an incremental instance with the get_updated_state function. 

## Expected Behavior
The airbyte documentation encourages the use of the IncrementalMixin class and considers the function "get_updated_state" as deprecated. Do you think to modify this incremental management, with the class IncrementalMixin?

### PR DESCRIPTION
**Pull request by bazarnov**
## What
Resolving:
- https://github.com/airbytehq/airbyte/issues/24697
- https://github.com/airbytehq/airbyte/issues/22929

## How
- fixed the bug when the `user_lifetime_insights` stream could return the records with missing `end_time` or `value` keys
- refactored the `STATE` management by migrating from the `get_updated_state()` method to `IncrementalMixin` inheritance
- covered both cases with `unit tests`

## 🚨 User Impact 🚨
No impact is expected - no breaking changes.

### CODE REVIEW

**Comment by octavia-squidington-iii** [comment]
##  source-instagram test report (commit [`e111160afd`](https://github.com/airbytehq/airbyte/pull/27908/commits/e111160afda03b37ff3a5acf6aa4a55a1e8cf883)) - ✅

⏲️  Total pipeline duration: 06mn24s 

| Step                                                                    | Result   |
|:------------------------------------------------------------------------|:---------|
| Validate airbyte-integrations/connectors/source-instagram/metadata.yaml | ✅        |
| Connector version semver check                                          | ✅        |
| Connector version increment check                                       | ✅        |
| QA checks                                                               | ✅        |
| Code format checks                                                      | ✅        |
| Connector package install                                               | ✅        |
| Build source-instagram docker image for platform linux/x86_64           | ✅        |
| Unit tests                                                              | ✅        |
| Integration tests                                                       | ✅        |
| Acceptance tests                                                        | ✅        |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/baz_source-instagram-fix-noo-end-time-users-insights/1688373276/e111160afda03b37ff3a5acf6aa4a55a1e8cf883/source-instagram/1.0.9/output.html)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/tools/ci_connector_ops/ci_connector_ops/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=source-instagram test
```

File: airbyte-integrations/connectors/source-instagram/source_instagram/streams.py [hunk 1]
```diff
@@ -218,8 +238,16 @@ def read_records(
                 f"temporarily unavailable data."
             )
             self.should_exit_gracefully = True
+
         yield from complete_records
 
+        # update state using IncrementalMixin
+        # reference issue: https://github.com/airbytehq/airbyte/issues/24697
+        if sync_mode == SyncMode.incremental:
+            if complete_records:
```
Comment (lazebnyi) [inline airbyte-integrations/connectors/source-instagram/source_instagram/streams.py:247]: better to use `AND` to increase if statements qty

**Review by lazebnyi — CHANGES_REQUESTED** [review]

File: airbyte-integrations/connectors/source-instagram/source_instagram/streams.py (hunk 1 above)
Comment (bazarnov) [reply airbyte-integrations/connectors/source-instagram/source_instagram/streams.py]: Done

**Review by lazebnyi — APPROVED** [review]

**Comment by octavia-squidington-iii** [comment]
##  source-instagram test report (commit [`4c0a428a14`](https://github.com/airbytehq/airbyte/pull/27908/commits/4c0a428a144924caeb9f8cc23bf5c1391691f59a)) - ✅

⏲️  Total pipeline duration: 07mn16s 

| Step                                                                    | Result   |
|:------------------------------------------------------------------------|:---------|
| Validate airbyte-integrations/connectors/source-instagram/metadata.yaml | ✅        |
| Connector version semver check                                          | ✅        |
| QA checks                                                               | ✅        |
| Code format checks                                                      | ✅        |
| Connector package install                                               | ✅        |
| Build source-instagram docker image for platform linux/x86_64           | ✅        |
| Unit tests                                                              | ✅        |
| Integration tests                                                       | ✅        |
| Acceptance tests                                                        | ✅        |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/baz_source-instagram-fix-noo-end-time-users-insights/1688568503/4c0a428a144924caeb9f8cc23bf5c1391691f59a/source-instagram/1.0.9/output.html)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/tools/ci_connector_ops/ci_connector_ops/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=source-instagram test
```

**Pull request merged**